pkg/params/params.go

```diff
@@ -301,6 +301,9 @@ func (p *Param) AsFloat64() float64 {
 }
 
 func (p *Param) AsInt() int {
+	if p == nil {
+		return 0
+	}
```

Comment:
I suppose we need to do it on all `As*` method. I'm wondering why somebody will call these on nil.

Reply:
It is the `top block-io` gadget:
```
	t.config.MaxRows = params.Get(gadgets.ParamMaxRows).AsInt()
	t.config.SortBy = params.Get(gadgets.ParamSortBy).AsStringSlice()
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6a118f]

goroutine 1379 [running]:
github.com/inspektor-gadget/inspektor-gadget/pkg/params.(*Param).AsInt(...)
	/home/alban/go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/params/params.go:307
github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/top/block-io/tracer.(*Tracer).Init(0xc0025ef400, {0x7f817ae384c0?, 0xc004656000?})
	/home/alban/go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/top/block-io/tracer/tracer.go:337 +0xaf
github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local.(*Runtime).RunGadget(0x1e28610?, {0x1e2d4d0, 0xc004656000})
	/home/alban/go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local/local.go:58 +0x20e
github.com/inspektor-gadget/inspektor-gadget/internal/benchmarks.BenchmarkAllGadgetsManyContainers.func2(0xc003d67680)
	/home/alban/go/src/github.com/inspektor-gadget/inspektor-gadget/internal/benchmarks/benchmarks_test.go:232 +0x677
testing.(*B).runN(0xc003d67680, 0x1)
	/home/alban/programs/golang/go/src/testing/benchmark.go:193 +0x102
testing.(*B).run1.func1()
	/home/alban/programs/golang/go/src/testing/benchmark.go:233 +0x59
created by testing.(*B).run1
	/home/alban/programs/golang/go/src/testing/benchmark.go:226 +0x9c
exit status 2
```

Reply:
I am fixing this in benchmarks_test.go by fetching the missing params from the gadget type (`top` in this case).

Thanks @flyth for the explanation